modules/t/geneTreeAdaptor.t

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env perl
+# See the NOTICE file distributed with this work for additional information
+# regarding copyright ownership.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+use strict;
+use warnings;
+
+use Test::More;
+use Test::Exception;
+
+use Bio::EnsEMBL::Test::MultiTestDB;
+use Bio::EnsEMBL::Test::TestUtils;
+
+use Bio::EnsEMBL::Utils::Exception qw (warning verbose);
+
+my $multi = Bio::EnsEMBL::Test::MultiTestDB->new( "homology" );
+my $compara_dba = $multi->get_DBAdaptor( "compara" );
+
+my $gene_tree_adaptor = $compara_dba->get_GeneTreeAdaptor();
+my $gene_member = $compara_dba->get_GeneMemberAdaptor->fetch_by_stable_id("ENSTSYG00000021671");
+my $gene_tree_other = $gene_tree_adaptor->fetch_default_for_Member($gene_member, "other");
+my $gene_tree_default = $gene_tree_adaptor->fetch_default_for_Member($gene_member, "default");
+
+subtest "Test Bio::EnsEMBL::Compara::DBSQL::GeneTreeAdaptor delete_tree method", sub {
+
+    my $default_before = $gene_member->has_GeneTree("default");
+    my $other_before = $gene_member->has_GeneTree("other");
+    is($default_before, 1, "Num default trees");
+    is($other_before, 1, "Num other trees");
+    $gene_tree_adaptor->delete_tree($gene_tree_default);
+    $gene_member = $compara_dba->get_GeneMemberAdaptor->fetch_by_stable_id("ENSTSYG00000021671");
+    my $default_after = $gene_member->has_GeneTree("default");
+    my $other_after = $gene_member->has_GeneTree("other");
+    is($default_after, 0, "Num default tree after delete");
+    is($other_after, $other_before, "Num other tree after default delete");
+
+    done_testing();
+};
+
```

Comment:
This test looks almost trivial, but it would have failed on the previous version of the `delete_tree` method, and provides additional confirmation that the method is now more robust even to the most unbargained-for blip.
```suggestion

my $gene_member2 = $compara_dba->get_GeneMemberAdaptor->fetch_by_stable_id("ENSXMAG00000006974");
my $gene_tree2_default = $gene_tree_adaptor->fetch_default_for_Member($gene_member2, "default");
my $all_trees = $gene_tree_adaptor->fetch_all_by_Member($gene_member2, -CLUSTERSET_ID => "phyml-aa");
my $gene_tree2_linked = $all_trees->[0];

subtest "Test Bio::EnsEMBL::Compara::DBSQL::GeneTreeAdaptor delete_tree method on linked protein tree", sub {

    my $default_before = $gene_member2->has_GeneTree("default");
    is($default_before, 1, "Default tree status before linked-tree delete");
    $gene_tree_adaptor->delete_tree($gene_tree2_linked);
    $gene_member2 = $compara_dba->get_GeneMemberAdaptor->fetch_by_stable_id("ENSXMAG00000006974");
    my $default_after = $gene_member2->has_GeneTree("default");
    is($default_after, 1, "Default tree status after linked-tree delete");

    done_testing();
};

```

Reply:
As above :+1: